### ISSUE TITLE
plugin-installer needs to be adapted for composer2

### ISSUE THREAD

**Issue by mstilkerich** [roundcube/plugin-installer#25]
Hello,

  composer v2 has now been released and v1 will be [considered EOL](https://blog.packagist.com/composer-2-0-is-now-available/) soon. Composer v2 has some [API changes for custom installers](https://getcomposer.org/upgrade/UPGRADE-2.0.md), specifically concerning this project the ```LibraryInstaller``` install/update/uninstall methods may now work asynchronously and return a PromiseInterface. There is an [example](https://github.com/composer/installers/commit/5006d0c28730ade233a8f42ec31ac68fb1c5c9bb) on how to adapt the code to be compatible with both composer v1 and v2.

As the composer project is encouraging all users to upgrade ASAP and indicates that they plan to cease support for v1, it would be great if plugin-installer could be made compatible with composer v2. I'm already receiving first issues (blind-coder/rcmcarddav#311) causing failure of roundcube plugin installation when composer v2 is used. I verified that adapting the code as in the example fixes the issue.

**Comment by thomascube** [comment]
Fixed in current master. Please verify the fix with `"roundcube/plugin-installer": "dev-master"` in your composer.json. It worked for me and once somebody confirms, we'll push the change as a new version.

**Comment by pgnd** [comment]
@thomascube 

```
composer -V
	Composer version 1.10.17 2020-10-30 22:31:58

composer self-update --2
composer -V
	Composer version 2.0.4 2020-10-30 22:39:11


cd /srv/webapps/roundcubemail/

	composer.json

		"require": {
			...
-			"roundcube/plugin-installer": "~0.1.6",
+			"roundcube/plugin-installer": "dev-master",
			...
		},

roundcube/plugin-installer": "dev-master"


composer  update --no-dev
	...
	Package operations: 0 installs, 1 update, 0 removals
	- Syncing roundcube/plugin-installer (dev-master dbe4cdb) into cache
	- Removing roundcube/plugin-installer (0.1.10)
	- Installing roundcube/plugin-installer (dev-master dbe4cdb): Cloning dbe4cdb4e9 from cache
	...

composer -nv remove roundcube/carddav sabre/vobject
rm -rf plugins/carddav

composer -nvv require roundcube/carddav
	Using version ^4.0 for roundcube/carddav
	./composer.json has been updated
	Running composer update roundcube/carddav
	Loading composer repositories with package information
	Updating dependencies
	Lock file operations: 12 installs, 0 updates, 0 removals
	- Locking guzzlehttp/guzzle (7.2.0)
	- Locking guzzlehttp/promises (1.4.0)
	- Locking guzzlehttp/psr7 (1.7.0)
	- Locking mstilkerich/carddavclient (v0.1.6)
	- Locking psr/http-client (1.0.1)
	- Locking psr/http-message (1.0.1)
	- Locking psr/log (1.1.3)
	- Locking ralouphie/getallheaders (3.0.3)
	- Locking roundcube/carddav (v4.0.1)
	- Locking sabre/uri (2.2.1)
	- Locking sabre/vobject (4.3.2)
	- Locking sabre/xml (2.2.3)
	Writing lock file
	Installing dependencies from lock file (including require-dev)
	Package operations: 12 installs, 0 updates, 0 removals
	- Installing psr/http-message (1.0.1): Extracting archive
	- Installing psr/http-client (1.0.1): Extracting archive
	- Installing ralouphie/getallheaders (3.0.3): Extracting archive
	- Installing guzzlehttp/psr7 (1.7.0): Extracting archive
	- Installing guzzlehttp/promises (1.4.0): Extracting archive
	- Installing guzzlehttp/guzzle (7.2.0): Extracting archive
	- Installing psr/log (1.1.3): Extracting archive
	- Installing sabre/uri (2.2.1): Extracting archive
	- Installing sabre/xml (2.2.3): Extracting archive
	- Installing sabre/vobject (4.3.2): Extracting archive
	- Installing mstilkerich/carddavclient (v0.1.6): Extracting archive
	- Installing roundcube/carddav (v4.0.1): Extracting archive
	0/2 [>---------------------------]   0%Creating plugin config file
	2 package suggestions were added by new dependencies, use `composer suggest` to see details.
	Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
	Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
	Generating autoload files
	5 packages you are using are looking for funding.
	Use the `composer fund` command to find out more!
```

composer v2 plugin install looks good

**Comment by mstilkerich** [comment]
Thanks for the fast fix, I can also confirm it works!

**Comment by olafnorge** [comment]
@thomascube thanks for fixing it. Would you mind releasing it also?

**Comment by thomascube** [comment]
> thanks for fixing it. Would you mind releasing it also?

@olafnorge It hat been released as version 0.2.0 via packagist but you might need to update your composer.json file.
See https://roundcube.net/news/2020/11/04/installer-update-for-composer-v2
and https://packagist.org/packages/roundcube/plugin-installer#0.2.0

**Comment by olafnorge** [comment]
Ohh, I missed that. Sorry for the noise and again thanks for pointing out.
